frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -174,6 +174,9 @@ function ReaderHighlight:init()
         return {
             text= _("Hyphenate"),
             show_in_highlight_dialog_func = function()
+                if not _self.ui.document or not _self.ui.document.provider or not _self.ui.document_provider ~= "crengine" then
+                    return false
+                end
                 return _self.ui.userhyph:isAvailable() and not _self.selected_text.text:find("[ ,;-%.\n]")
```

Comment:
No need for such complicated test.
Just check for the existence of `_self.ui.userhyph`:

`return _self.ui.userhyph and _self.ui.userhyph:isAvailable() and not _self.selected_text.text:find("[ ,;-%.\n]")`